### ISSUE TITLE
Best way to customize this repo to make it work with custom chain?

### ISSUE THREAD

**Issue by xlc** [polkadot-js/apps#511]
I am using this repo to interact with my custom substrate chain however there are few pain points to make it work.

1. I am not able to find the best place to inject my custom types into the repo. I am currently just throw this code to some index.ts of the forked repo. It will be good if there are some ways to make this cleaner.

```
import typeRegistry from '@polkadot/types/codec/typeRegistry';
import * as myTypes from 'myTypes';

typeRegistry.register(myTypes);
```

2. No easy way to customize the endpoint options. Basically I want to modify this file https://github.com/polkadot-js/ui/blob/master/packages/ui-settings/src/defaults.ts and make it available. However I don't want to fork both apps repo and ui repo and change apps repo to use the forked ui repo and keep both repos up to date. It is just too much hassle.
It will be good to provide some way to inject the default values from apps repo, or environment value, or some config file. So that I can minimize the changes required to the forked repo. To none if possible (i.e. able to customize the values outside the repo).

**Comment by jacogr** [comment]
Agreed. I've been mulling this around in my head since you added the types injection - I believe the best approach (or at least the one I though of) is to extend the setting apps to allow upload of the custom definitions. So basically, I was thinking along the lines of -

- types upload inside app-settings
- save these to the local storage
- load these are part of chain init

The only problematic part here would be if there are non default types, i.e. you created your own classes.

For the endpoint options, these can actually be retrieved from the chain now. (It however has not been integrated yet). So basically we need to do an additional query in the init to chain properties vai https://github.com/polkadot-js/api/blob/master/packages/type-jsonrpc/src/system.ts#L29 and can then inject these.

Basically if you take a look at the bbq.json file in Substrate, is has a single property - it is basically just JSON so we can add anything here.

Sorry we are lagging a bit behind with support of both of these :(

**Comment by jacogr** [comment]
So for all the properties, this has been merged into Substrate - https://github.com/paritytech/substrate/pull/1286/files allowing for `tokenSymbol`, `tokenDecimals` and `networkId` - the UI supports these as named above and will read it from the chain. (So part of the issue logged can be configured now at least)
